services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -365,14 +367,22 @@ public News unScheduleNews(News news, org.exoplatform.services.security.Identity
    * {@inheritDoc}
    */
   public News postNews(News news, String poster) throws Exception {
-    postNewsActivity(news);
-    updateNews(news, poster, null, news.isPublished());
-    sendNotification(poster, news, NotificationConstants.NOTIFICATION_CONTEXT.POST_NEWS);
-    if (news.isPublished()) {
-      publishNews(news, poster);
-    }
-    NewsUtils.broadcastEvent(NewsUtils.POST_NEWS_ARTICLE, news.getId(), news);//Gamification
-    NewsUtils.broadcastEvent(NewsUtils.POST_NEWS, news.getAuthor(), news);//Analytics
+    News existNews;
+    try {
+      existNews = newsStorage.getNewsById(news.getId(), false);
+    } catch (ItemNotFoundException e) {
```

Comment:
as getNewsById throws a new Exception, are you sure that the type of the exception is ItemNotFound in this catch ?

Reply:
Yes i'm sure